src/run/run_step.go

```diff
@@ -150,46 +150,51 @@ func run(ctx context.Context, state *core.BuildState, label core.AnnotatedOutput
 
 	// ReplaceSequences always quotes stuff in case it contains spaces or special characters,
 	// that works fine if we interpret it as a shell but not to pass it as an argument here.
-	command := ""
 	switch {
+	case overrideCmd != "":
+		command, _ := core.ReplaceSequences(state, target, overrideCmd)
+		// We don't care about passed in args when an override command is provided
+		args = process.BashCommand("bash", strings.Trim(command, "\""), true)
 	case label.Annotation != "":
 		entryPoint, ok := target.EntryPoints[label.Annotation]
 		if !ok {
 			log.Fatalf("Cannot run %s as has no entry point %s", label, label.Annotation)
 		}
+		var command string
 		if tmpDir {
 			command = filepath.Join(dir, entryPoint)
 		} else {
 			command = filepath.Join(target.OutDir(), entryPoint)
 		}
-	case overrideCmd != "":
-		command = overrideCmd
+		args = append(strings.Split(command, " "), args...)
 	default:
 		// out_exe handles java binary stuff by invoking the .jar with java as necessary
+		var command string
 		if tmpDir {
 			command = filepath.Join(dir, target.Outputs()[0])
 		} else {
 			command, _ = core.ReplaceSequences(state, target, fmt.Sprintf("$(out_exe %s)", target.Label))
+			command = strings.Trim(command, "\"")
 		}
+		args = append(strings.Split(command, " "), args...)
 	}
-	arg0 := strings.Trim(command, "\"")
+
 	// Handle targets where $(exe ...) returns something nontrivial
-	splitCmd := strings.Split(arg0, " ")
-	if !strings.Contains(splitCmd[0], "/") {
+	if !strings.Contains(args[0], "/") {
```

Comment:
We could use this instead: https://golang.org/pkg/path/filepath/#IsAbs

Reply:
The block wouldn't be executed for relative paths with `/`, would that be a problem?

Reply:
Nah this is checking `args[0]` only which is the same as `filepath.IsAbs`